src/tools/r2rdump/R2RSignature.cs

```diff
@@ -1072,6 +1077,24 @@ private void ParseType(StringBuilder builder)
                     throw new NotImplementedException();
             }
         }
+
+        public MetadataReader GetMetadataReaderFromModuleOverride()
+        {
+            if (PeekElementType() == CorElementType.ELEMENT_TYPE_MODULE_ZAPSIG)
+            {
+                var currentOffset = _offset;
+
+                ReadElementType();
+                int moduleIndex = (int)ReadUInt();
+                EcmaMetadataReader refAsmReader = _contextReader.OpenReferenceAssembly(moduleIndex);
+
+                _offset = currentOffset;
+
+                return refAsmReader.MetadataReader;
+            }
+            return null;
+        }
+
```

Comment:
The R2RDump changes look like general goodness - should we merge them into master?

Reply:
They are already in master. I cherry-picked the commit from master and made it part of this PR.
This PR is really just the changes in prestub.cpp

Reply:
I see, great, thanks for clarification!